sdktests/server_side_persistence_read_write.go

```diff
@@ -0,0 +1,469 @@
+	})
+
+	t.Run("ttl cache", func(t *ldtest.T) {
+		t.Run("ignores dropped flags", s.ignoresDroppedFlagsWithTTLCache)
+	})
+}
+
+func (s *ServerSidePersistentTests) initializesStoreWhenDataReceived(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
+	})
+
+	sdkData := s.makeSDKDataWithFlag("flag-key", 1, ldvalue.String("value"))
+	_, configurers := s.setupDataSources(t, sdkData)
+	configurers = append(configurers, persistence)
+
+	_, err := s.persistentStore.ReadField("launchdarkly:$inited")
+	require.Error(t, err) // should not exist
+
+	_ = NewSDKClient(t, s.baseSDKConfigurationPlus(configurers...)...)
+	s.eventuallyRequireDataStoreInit(t, "launchdarkly")
+}
+
+func (s *ServerSidePersistentTests) appliesUpdatesToStore(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
+	})
+
+	sdkData := s.makeSDKDataWithFlag("flag-key", 1, ldvalue.String("value"))
+	stream, configurers := s.setupDataSources(t, sdkData)
+	configurers = append(configurers, persistence)
+
+	_, err := s.persistentStore.ReadField("launchdarkly:$inited")
+	require.Error(t, err) // should not exist
+
+	_ = NewSDKClient(t, s.baseSDKConfigurationPlus(configurers...)...)
+	s.eventuallyRequireDataStoreInit(t, "launchdarkly")
+	s.eventuallyValidateFlagData(t, "launchdarkly", map[string]m.Matcher{
+		"flag-key": basicFlagValidationMatcher("flag-key", 1, "value"),
+	})
+
+	updateData := s.makeFlagData("flag-key", 2, ldvalue.String("new-value"))
+	stream.StreamingService().PushUpdate("flags", "flag-key", updateData)
+	s.eventuallyValidateFlagData(t, "launchdarkly", map[string]m.Matcher{
+		"flag-key": basicFlagValidationMatcher("flag-key", 2, "new-value"),
+	})
+}
+
+func (s *ServerSidePersistentTests) dataSourceUpdatesRespectVersioning(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
+	})
+
+	sdkData := s.makeSDKDataWithFlag("flag-key", 1, ldvalue.String("value"))
+	stream, configurers := s.setupDataSources(t, sdkData)
+	configurers = append(configurers, persistence)
+
+	_ = NewSDKClient(t, s.baseSDKConfigurationPlus(configurers...)...)
+	s.eventuallyRequireDataStoreInit(t, "launchdarkly")
+
+	require.NoError(t, s.persistentStore.WriteData("launchdarkly:features", s.initialFlags))
```

Comment:
I think the relationship between the flag versions could be made clearer here. We don't see the setup of `initialFlags` in the test, but we are expecting that `1` is lower and `200` is higher. 

Could the lower/higher be derived from the initial data, or could the initial data be generated by a function call that passes in versions?